src/components/Recipes/AddToMealplanContainer.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react'
+import { AddRecipeTitleToMealplanToggle } from '../Button'
+import imagePathPlus from '../../images/inverted-plus.svg'
+import { v4 as uuidv4 } from 'uuid'
+
+import styled from 'styled-components'
+import { saveToLocal, loadFromLocal } from '../../utils/localStorage'
+
+export default function AddToMealplanContainer({ title }) {
+  const [active, setActive] = useState(() => {
+    const mealplanList = loadFromLocal('mealplanList') || {
+      title: '',
+      items: [{ id: 'initial', item: '', done: false }],
+    }
+
+    const initialState =
+      mealplanList.items.find((meal) => {
+        return meal.item === title
+      }) !== undefined
+    return initialState
+  })
+
+  const addRecipeToMealplan = (title) => {
+    const newMeal = { id: uuidv4(), item: title, done: false }
+    const mealplanList = loadFromLocal('mealplanList') || {
+      title: '',
+      items: [{ id: 'initial', item: '', done: false }],
+    }
+    const newMealplanList = {
+      ...mealplanList,
+      items:
+        mealplanList.items.length === 0 ||
+        (mealplanList.items.length === 1 && mealplanList.items[0].item === '')
```

Comment:
Brauchst du die Abfrage `mealplanList.items.length === 1 ` hier noch, wenn du in 32 schon ausschließt, dass wenn die Länge 0 ist, dass nichts passiert?

Reply:
Das ist ein bisschen kompliziert, weil ich beim ersten Rendern meines Mealplans immer schon ein leeres MealplanItem in meiner Liste rendern lasse. Das passiert aber nicht, wenn ich in meiner App direkt von der RecipePage einsteige. Dann hat meine Liste noch kein Element. Daher kann bei noch keinem vorhanden Element das neue einfach hinzugefügt werden. Wenn mein items-array aber length 1 hat, das MealplanItem aber einen leeren String als value hat (wie bei initial render meiner App vorgesehen), dann wird das leere Element durch das neue meal ersetzt. Sonst habe ich immer ein leeres Element ganz oben in meiner Liste, dass ich dann händisch löschen müsste.